### DISCUSSION TITLE
Visualizing Transport Connections - Showcase

### DISCUSSION THREAD

**Discussion by avarga** [inet-framework/inet#597] [Showcases]
Discuss the "Visualizing Transport Connections" showcase here. Ideas and recommendations are welcome.

**Comment by ManiAm** [comment]
**Suggestion:** This module can be renamed to 'Visualizing Transport End points' rather than Transport Connection which is vague and could refer to the path too. Transport Protocols are end to end protocols (rather than hop-to-hop) and marking the end of each connection is exactly what this visualizer is doing.

**Comment by levy** [comment]
Thanks for the suggestion!
Indeed, this visualizer only displays the connection end points right now, but its intended purpose is more than that. Although this may not be clear from its barely existing documentation, we have our limits.

The connection visualizer could display the connection path if there is one which can be figured out by analyzing the network state at all. In this case, what I mean by the connection path is not to be confused with the recent activity (i.e. packets being sent along a path) but rather some allocated network resource (e.g. routing table entries). This is not implemented because it's not so simple do it, some network features (e.g. load balancing) make displaying the connection path even more complicated. Moreover, if we were to split the functionality into two visualizers, then we would end up maintaining the set of open connections twice.
